sdks/java/core/src/test/java/org/apache/beam/sdk/metrics/MetricsTest.java

```diff
@@ -95,4 +107,54 @@ public void counterToCell() {
     counter.dec();
     assertThat(cell.getCumulative(), CoreMatchers.equalTo(42L));
   }
+
+  @Category({RunnableOnService.class, UsesMetrics.class})
+  @Test
+  public void metricsE2E() {
+    final Counter count = Metrics.counter(MetricsTest.class, "count");
+    Pipeline pipeline = TestPipeline.create();
+    pipeline
+        .apply(Create.of(5, 8, 13))
+        .apply("MyStep1", ParDo.of(new DoFn<Integer, Integer>() {
+          @ProcessElement
+          public void processElement(ProcessContext c) {
+            Distribution values = Metrics.distribution(MetricsTest.class, "input");
+            count.inc();
+            values.update(c.element());
+
+            c.output(c.element());
+            c.output(c.element());
+          }
+        }))
+        .apply("MyStep2", ParDo.of(new DoFn<Integer, Integer>() {
+          @ProcessElement
+          public void processElement(ProcessContext c) {
+            Distribution values = Metrics.distribution(MetricsTest.class, "input");
+            count.inc();
+            values.update(c.element());
+          }
+        }));
+    PipelineResult result = pipeline.run();
+
+    result.waitUntilFinish();
+
+    MetricQueryResults metrics = result.metrics().queryMetrics(MetricsFilter.builder()
+      .addNameFilter(MetricNameFilter.inNamespace(MetricsTest.class))
+      .build());
+    final String step1Name = "MyStep1/AnonymousParDo/AnonymousParMultiDo";
```

Comment:
Yeah, my guess would be just changing this to the name contains "MyStep1" would be the best.